lib/xandra/protocol.ex

```diff
@@ -542,10 +542,11 @@ defmodule Xandra.Protocol do
   end
 
   # Prepared
-  defp decode_result_response(<<0x0004::32-signed, buffer::bits>>, %Prepared{} = prepared, _options) do
+  defp decode_result_response(<<0x0004::32-signed, buffer::bits>>, %Prepared{} = prepared, options) do
+    atom_keys? = Keyword.fetch!(options, :atom_keys?)
     decode_string(id <- buffer)
-    {%{columns: bound_columns}, buffer} = decode_metadata(buffer, %Page{})
-    {%{columns: result_columns}, <<>>} = decode_metadata(buffer, %Page{})
+    {%{columns: bound_columns}, buffer} = decode_metadata(buffer, %Page{}, atom_keys?)
```

Comment:
With this change we enforce that params for prepared query should have atom keys:
```elixir
prepared = Xandra.prepare!(conn, statement)
result = Xandra.execute!(conn, prepared, %{figure: 123})
assert Enum.to_list(result) == [%{figure: 123}]
```
I think it makes sense, right? Or the `:atom_keys` option should affect only result columns? What do you think @scohen @whatyouhide?

Reply:
Agreed, I think having consistency in the keys is correct. 
Are you saying that I need to update the docs to reflect this fact as well? I'm sorry that I missed it if so, it's been a long time since I've been deep in this PR.